src/main/java/org/mockito/listeners/VerificationListener.java

```diff
@@ -0,0 +1,24 @@
+package org.mockito.listeners;
+
+import org.mockito.verification.VerificationData;
+import org.mockito.verification.VerificationMode;
+import org.mockito.verification.VerificationSucceededEvent;
+
+/**
+ * This listener can be notified of verify invocations on a mock.
+ *
+ * For this to happen, it must be registered using {@link org.mockito.internal.progress.MockingProgress#addListener(MockitoListener)}.
+ */
+public interface VerificationListener extends MockitoListener {
+    /**
+     * Called when a verification completes successfully without throwing an exception.
+     *
+     * @param verificationSucceededEvent contains all information about the event.
+     *                                   Consisting of the mock that was verified on,
+     *                                   the {@link org.mockito.verification.VerificationMode} used
+     *                                   and the {@link VerificationData}
+     */
+    void onVerificationSucceeded(VerificationSucceededEvent verificationSucceededEvent);
+
+    void onVerificationException(Object mock, VerificationMode actualMode, Throwable failure);
```

Comment:
I would prefer listener interfaces with only one method (FunctionalInterface), this way lambda-expressions can be used as listeners too.